docs/content/dev/notes.md

```diff
@@ -0,0 +1,58 @@
+# Notes
+
+Each note in HedgeDoc 2 contains the following information:
+
+- publicId (`b604x5885k9k01bq7tsmawvnp0`)
+- a list of aliases (`[hedgedoc-2, hedgedoc-next]`)
+- groupPermissions
+- userPermissions
+- viewCount (`0`)
+- owner
+- revisions
+- authorColors
+- historyEntries
+- description (`All you never wanted to know about notes`)
+- title (`Notes`)
+- tags (`[features, cool, update]`)
+- version
+
+The `publicId` is the default possibility of identifying a note. It will be a randomly generated 128-bit value encoded with [base32-encode](https://www.npmjs.com/package/base32-encode) using the crockford variant and converted to lowercase. This variant of base32 is used, because that results in ids that only use one case of alpha characters, numeric characters and other url safe characters. We convert the id to lowercase, because we want to minimize case confusion.
+
+`aliases` are the other way of identifying a note. There can be any number of them, and the owner of the note is able to add or remove them. All aliases are just strings (especially to accommodate the old identifier from HedgeDoc 1 [see below](#conversion-of-hedgedoc-1-notes)), but new aliases added with HedgeDoc 2 will only allow characters matching this regex: `[a-z0-9\-_]`. This is done to once again prevent case confusion.
+
+`groupPermissions` and `userPermissions` each hold a list of the appropriate permissions.
+Each permission holds a reference to a note and a user/group and specify what the user/group is allowed to do.
+Each permission is additive, that means a user that has only the right to read a note via a group, but the right to write via a different group or directly for his user, is able to write in the note.
+
+The `viewCount` is a simple counter that holds how often the note in question was requested.
+
+`owner` is the user that created the note or later got ownership of the note. The current owner is able to change the owner of the note to someone else.
```

Comment:
Maybe also add that only the owner has the ability to delete the note or to purge the revisions?

Reply:
added